compiler/noirc_frontend/src/elaborator/enums.rs

```diff
@@ -795,7 +795,7 @@ impl Elaborator<'_> {
 
         let value = match constant {
             Value::Bool(value) => SignedField::positive(value),
-            Value::Field(value) => SignedField::positive(value),
```

Comment:
This was a bug. Because `SignedField` is `impl Into<FieldElement>` it was transformed into a `FieldElement`, then that was sent to `positive`. The problem with that is that if we have a `SignedField(1, negative: true)` it's going to be passed as the largest field value into positive, effectively changing the meaning of that value (it's the same value for field elements but it could be a different one if we then need to convert that to another type).

That's why I removed `impl From<SignedField> for FieldElement`, which wasn't used in a lot of places (and most of the places it was used it was incorrect like here).